File: python/calculator/calculator_test.py
"""Pytest tests for the calculator sample (run via bazel test)."""

import pytest

from python.calculator.calculator import add


@pytest.mark.parametrize(
    ("a", "b", "expected"),
    [(2, 3, 5), (-1, 1, 0), (0, 0, 0), (-4, -6, -10)],
)
def test_add(a: int, b: int, expected: int) -> None:
    assert add(a, b) == expected


if __name__ == "__main__":
    raise SystemExit(pytest.main([__file__, "-v"]))

File: pkg/calculator/calculator_test.go
package calculator

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestAdd(t *testing.T) {
	tests := []struct {
		name       string
		a, b, want int
	}{
		{"positive", 2, 3, 5},
		{"mixed", -1, 1, 0},
		{"zero", 0, 0, 0},
		{"negative", -4, -6, -10},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			assert.Equal(t, tt.want, Add(tt.a, tt.b))
		})
	}
}
